include/pmacc/dimensions/DataSpaceOperations.hpp

```diff
@@ -266,27 +266,31 @@ namespace pmacc
         template<class TVEC>
         static HDINLINE DataSpace<DIM3> map(uint32_t pos)
         {
-            return DataSpace< DIM3 >(
-                pos                                      % TVEC::x::value,
-                pos /   TVEC::x::value                   % TVEC::y::value,
-                pos / ( TVEC::x::value * TVEC::y::value )
-            );
+            constexpr auto xyPlane = TVEC::x::value * TVEC::y::value;
```

Comment:
If the suggestion above did change so much, it could have been due to the parameter staying const.
Maybe try to use a temporary auto const var for this manipulation here as well?

Reply:
I do not get the point. This is already compile time.

Reply:
Dang, my app selected the wrong line. I Mena the 3D part where you do:

pos -=...

Maybe use a temporary auto const var there as well?

Reply:
This suggestion is not changing everything

Reply:
Ok, thanks for checking!

Reply:
Ok, thanks for checking!

Reply:
(you did test a 3D KHI, did you?)

Reply:
I did only small test and checked the PNGs, our cluster is fully blocked for days :-(

Reply:
But you need to compile a 3D case to see that code section in effect for registers

Reply:
Yes I compiled the 3D KHI, in the meanwhile I also run a default heating test on 4GPUs. The changes passed the test and I added it to the initial PR comment.